Protacon.RxMq.AzureServiceBus/Topic/AzureTopicSubscriber.cs

```diff
@@ -93,13 +94,17 @@ private void UpdateRules(SubscriptionClient subscriptionClient, AzureBusTopicSet
                     .ForEach(x => subscriptionClient.AddRuleAsync(x.Key, x.Value).Wait());
             }
 
-            private static T AsObject(string body)
+            private static T AsObject(string body, long arrival, bool addArrival = false)
             {
                 var parsed = JObject.Parse(body);
 
                 if (parsed["data"] == null)
                     throw new InvalidOperationException("Library expects data wrapped as { data: { ... } }");
 
+                if (addArrival) {
+                    parsed["data"]["arrival"] = arrival;
```

Comment:
Mietin olisko vähän tikemmin rajattu nimi parempi, ettei vahingossa ylikirjoita viestin "natiivia" proppia. Esim
```suggestion
                    parsed["data"]["rxmqarrival"] = arrival;
```

Reply:
Vois olla. Täytyy vaan katsoa miten toi kääntyy luettavaan formaattiin kun tosta datasta muodostetaan luokka-instansseja.